Enable more common compression filters for restoring

alimirjamali opened the pull request:
Since some users want pigz, zstd and some other compressions which might not be installed by default, enable them if they are installed.

Related: https://github.com/QubesOS/qubes-issues/issues/8291
Related: https://github.com/QubesOS/qubes-issues/issues/8211

alimirjamali committed:
Enable more common compression filters for restoring

Since some users want pigz, zstd and some other compressions which might
not be installed by default, enable them if they are installed.

Related: https://github.com/QubesOS/qubes-issues/issues/8291
Related: https://github.com/QubesOS/qubes-issues/issues/8211

marmarek commented on qubesadmin/tests/backup/backupcompatibility.py:
```diff
@@ -1563,13 +1563,19 @@ def test_210_r2(self):
             mock.patch(
                 'qubesadmin.backup.restore.BackupRestore._handle_appmenus_list',
-                functools.partial(self.mock_appmenus, qubesd_calls_queue)),
+                staticmethod(
```
Adding staticmethod don't look related to anything described in the commit message - should be a separate commit with info what/why.

alimirjamali replied:
I split the PR to two individual PRs. The 2nd commit is to reduce the clutter in unittest output. As Python asks the `functools.partial` to be wrapped within `staticmethod`.

alimirjamali committed:
Cleanup Backup Compatibility test output

Wrap `functools.partial` inside `staticmethod` to reduce test report
clutter. Fixing a small typo
